app/scripts/metamask-controller.js

```diff
@@ -2409,6 +2412,23 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Checks if the multichain accounts feature is enabled for state 2.
+   *
+   * @returns boolean - True if the feature is enabled for state 2, false otherwise.
+   */
+  isMultichainAccountsState2Enabled() {
+    const remoteFlag =
+      this.remoteFeatureFlagController?.state?.remoteFeatureFlags
+        ?.enableMultichainAccounts;
+    if (!remoteFlag?.enabled || remoteFlag.featureVersion !== '2') {
+      return false;
+    }
+    const min = remoteFlag.minimumVersion;
+    const current = process.env.METAMASK_VERSION;
+    return !min || semver.gte(current, min);
```

Comment:
We should rely on the existing helpers for this:
- https://github.com/MetaMask/metamask-extension/blob/main/ui/selectors/multichain-accounts/feature-flags.ts